engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/multiinstance/AbstractMultiInstanceBodyHandler.java

```diff
@@ -17,20 +17,29 @@
 import io.zeebe.msgpack.jsonpath.JsonPathQuery;
 import io.zeebe.msgpack.query.MsgPackQueryProcessor;
 import io.zeebe.msgpack.spec.MsgPackHelper;
+import io.zeebe.msgpack.spec.MsgPackWriter;
 import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import io.zeebe.util.buffer.BufferUtil;
 import java.util.Collections;
 import java.util.function.Function;
 import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
 import org.agrona.concurrent.UnsafeBuffer;
 
 public abstract class AbstractMultiInstanceBodyHandler
     extends AbstractHandler<ExecutableMultiInstanceBody> {
 
   private static final DirectBuffer NIL_VALUE = new UnsafeBuffer(MsgPackHelper.NIL);
+  private static final DirectBuffer LOOP_COUNTER_VARIABLE = BufferUtil.wrapString("loopCounter");
 
   private final Function<BpmnStep, BpmnStepHandler> innerHandlerLookup;
   private final MsgPackQueryProcessor queryProcessor = new MsgPackQueryProcessor();
 
+  private final MutableDirectBuffer loopCounterVariableBuffer =
+      new UnsafeBuffer(new byte[Long.BYTES + 1]);
```

Comment:
Is `+ 1` because MsgPack writes a "header" byte before?

Reply:
Correct.